src/Protocol/Protocol_1_9.cpp

```diff
@@ -1724,9 +1724,8 @@ bool cProtocol_1_9_0::CompressPacket(const AString & a_Packet, AString & a_Compr
 
 int cProtocol_1_9_0::GetParticleID(const AString & a_ParticleName)
 {
-	static bool IsInitialized = false;
 	static std::map<AString, int> ParticleMap;
-	if (!IsInitialized)
```

Comment:
A `static` variable is somewhere between a local variable (it is only accessible in the block where it is declared) and a global variable (it has one value, which stays between multiple entrances into that block). Basically, imagine this as being declared as a global variable `bool cProtocol_1_9_0::GetParticleID::Scope1::IsInitialized`. So it is only accessible in this `GetParticleID()` function, but it keeps its value from one function call to another.

Here's some code to illustrate:
```cpp
void printCounter(void)
{
  static int counter = 0;
  printf("%d\n", counter);
  counter += 1;
}

void printSequence(void)
{
  printCounter();  // Prints "0", "counter" is now 1
  printCounter();  // Prints "1", "counter" is now 2
  printCounter();  // Prints "2", "counter" is now 3
  counter = 4;  // compile error, "counter" not accessible from here
}
```

The primary purpose was supposed to be to make the `ParticleMap` initialization happen only once. The same effect can be obtained by asking the map if it's empty.